### ISSUE TITLE
Implement relational operations <=, ==, >=, <, >

### ISSUE THREAD

**Issue by certik** [symengine/symengine#1087]
For example, this:

```
 auto formula = parse("((x>=y) & (y>=z))*sign(x-y)")
```

Would work like the following:

```
 result = formula.lambda(x=3, y=2.5, z=1.5) = ((1.0)&(1.0))*1.0 = 1.0
 result = formula.lambda(x=3, y=3.5, z=1.5) = ((0.0)&(1.0))*1.0 = 0.0
```

In SymPy, you can do things like:

```
In [1]: x <= y
Out[1]: x ≤ y
```

**Comment by isuruf** [comment]
You can do this with `Piecewise`
`Piecewise( ((x-y in [0, inf)) and (y-z in [0, inf)), (x-y)/abs(x-y)), (True, 0))`

**Comment by bjodah** [comment]
FWIW, relationals in symengine would help a use-case I just encountered for `Lambdify`:

>>> cb = symengine.Lambdify([x, y], [x>=0, y>=0])
...
SympifyError: sympy2symengine: Cannot convert 'x >= 0' to a symengine type.
```
Where I would like to have this eventually:
```
>>> np.allclose(cb([2, -1]), [1.0, 0.0])
True
```

**Comment by isuruf** [comment]
See symengine/symengine.py#90. In it I used intervals (which was already implemented) and `contains`. `to_contains` can convert `x >= 0` to a symengine expression.
Now that And, Or, Not are all there, after implementing relationals, we don't need `contains` at all.

**Comment by ShikharJ** [comment]
Fixed via #1276.
